### DISCUSSION TITLE
CRA and Webpack5 for microfrontend

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [webpack/webpack#13133] [Q&A]
Hello. Please tell me. I am trying to make sure that CRA remains in the project as a development tool with an excellent DX. 
I added an alias for two versions of the webpack to the package
```
"webpack": "4.44.2",
    "webpack-5.32.0": "npm:webpack@5.32.0",
    "webpack-cli": "4.6.0"
```
 but I catch an error
` [webpack-cli] TypeError: Cannot read property 'includes' of undefined`

What can i do without create custom config and kill DX?

**Comment by alexander-akait** [comment]
CRA is not compatibility with webpack v5 yet

**Reply by popuguytheparrot** [reply]
i dont try use webpack5 in CRA. I create alias and webpack.config for build app. For development using CRA

**Reply by snitin315** [reply]
How did you run webpack?

**Reply by popuguytheparrot** [reply]
"build:mf": "node ./node_modules/webpack-5.32.0/bin/webpack.js --config ./webpack.config.js",

**Comment by snitin315** [comment]
As per your script in https://github.com/webpack/webpack/discussions/13133#discussioncomment-609187 you used `webpack-5.32.0` which is not yet compatible.

Also, we have a `WEBPACK_PACKAGE` env variable for using a custom webpack package.

```
     "webpack": "4.44.2",
     "webpack-5.32.0": "npm:webpack@5.32.0",
     "webpack-cli": "4.6.0"
```

`"build:mf": "webpack --config ./webpack.config.js"` - run with webpack 4

`"build:mf": "WEBPACK_PACKAGE=webpack-5.32.0 webpack --config ./webpack.config.js"` - run with webpack 5


Refer https://webpack.js.org/api/cli/#cli-environment-variables

**Reply by popuguytheparrot** [reply]
Will try

**Reply by popuguytheparrot** [reply]
So... Working version. Add only the required loaders 😊 
Please tell me about Terser and code compression. Do also need to connect everything? How to deal with production build?
